buildpack.md

```diff
@@ -245,19 +245,20 @@ The `/bin/detect` executable in each buildpack, when executed:
 - MAY read the app directory.
 - MAY read the detect environment as defined in the [Environment](#environment) section.
 - MAY emit error, warning, or debug messages to `stderr`.
-- MAY receive a TOML-formatted [Build Plan](#build-plan-toml) on `stdin`.
-- MAY contribute to the Build Plan by writing TOML to `<plan>`.
+- MAY augment the Build Plan by writing TOML to `<plan>`.
 - MUST set an exit status code as described in the [Buildpack Interface](#buildpack-interface) section.
 
-For each `/bin/detect`, the Build Plan received on `stdin` MUST be a map derived from the combined Build Plan contributions of all previous `/bin/detect` executables.
-In order to make contributions to the Build Plan, a `/bin/detect` executable MUST write entries to `<plan>` as top-level, TOML-formatted objects.
+In order to make contributions to the Build Plan, a `/bin/detect` executable MUST write entries to `<plan>` in two sections: `requires` and `provides`.
+Each section MUST be a list of entries formatted as described in the [Build Plan](#build-plan-toml) format section.
 
-The lifecycle MUST construct this map such that the top-level values from later buildpacks override the entire top-level values from earlier buildpacks.
-The lifecycle MUST NOT include any changes in this map that are contributed by optional buildpacks that returned non-zero exit statuses.
-The final Build Plan is the fully-merged map that includes the contributions of the final `/bin/detect` executable.
+For a given buildpack group,
+- If a required buildpack provides a dependency that is not required by the same buildpack or a subsequent buildpack, the group MUST fail to detect.
+- If a required buildpack requires a dependency that is not provided by the same buildpack or a previous buildpack, the group MUST fail to detect.
+- If an optional buildpack provides a dependency that is not required by the same buildpack or a subsequent buildpack, it MUST be excluded from the build phase and its requires and provides MUST be excluded from the Build Plan.
```

Comment:
can we clarify that "it" (as in "it MUST be excluded") is the optional buildpack, and not the thing that was provided?

Reply:
Done :)